prettyPrinters/lldb_providers.py

```diff
@@ -154,6 +155,25 @@ def StdStrSummaryProvider(valobj, _dict):
     return '"%s"' % data
 
 
+def StdFFIStrSummaryProvider(valobj, _dict, is_null_terminated=False):
+    # type: (SBValue, dict, bool) -> str
+    process = valobj.GetProcess()
+    error = SBError()
+    slice_ptr = valobj.GetLoadAddress()
+    if slice_ptr == LLDB_INVALID_ADDRESS:
+        return ""
+    char_ptr_type = valobj.GetTarget().GetBasicType(eBasicTypeChar).GetPointerType()
+    start = valobj.CreateValueFromAddress('start', slice_ptr, char_ptr_type).GetValueAsUnsigned()
+    length = process.ReadPointerFromMemory(slice_ptr + process.GetAddressByteSize(), error)
+    if is_null_terminated:
+        length = length - 1
+    if length == 0:
+        return '""'
+    data = process.ReadMemory(start, length, error)
+    data = data.decode(encoding='UTF-8') if PY3 else data
```

Comment:
Hmm, what if I have `CString` on Windows with `UTF-16` encoding? Does CLion do something special for C string in C/C++?

Reply:
I'm not sure `CString` should be used as a `UTF-16`-encoded string. I found a [discussion](https://internals.rust-lang.org/t/pre-rfc-deprecate-and-replace-cstr-cstring/5016) on this topic, and I think [this comment](https://internals.rust-lang.org/t/pre-rfc-deprecate-and-replace-cstr-cstring/5016/38) is probably the most informative.

As for C/C++, I haven't found special handling of UTF-16 C strings in CLion.